packages/cli-core/src/package-manager-controller/package_manager_controller.ts

```diff
@@ -112,6 +120,45 @@ Get started by running \`${this.binaryRunner} amplify sandbox\`.`;
     );
   }
 
+  /**
+   * runWithPackageManager - Factory function that runs a command with the specified package manager's binary runner
+   */
+  runWithPackageManager(
+    args: string[] = [],
+    dir: string,
+    options?: {
+      env?: Record<string, string>;
+      stdin?: 'inherit' | 'pipe' | 'ignore';
+      stdout?: 'inherit' | 'pipe' | 'ignore';
+      stderr?: 'inherit' | 'pipe' | 'ignore';
+      extendEnv?: boolean;
+    }
+  ): ExecaChildProcess<string> {
+    return this.executeWithDebugLogger(
+      dir,
+      this.binaryRunner,
+      args,
+      this.execa,
+      options
+    );
+  }
+
+  getPackageManagerCommandArgs = (
+    invokableCommand: InvokableCommand,
+    backendLocator: BackendLocator
+  ) => [
+    'cdk',
+    invokableCommand.toString(),
+    // This is unfortunate. CDK writes everything to stderr without `--ci` flag and we need to differentiate between the two.
+    // See https://github.com/aws/aws-cdk/issues/7717 for more details.
+    '--ci',
+    '--app',
+    `'${this.binaryRunner} tsx ${backendLocator.locate()}'`,
+    '--all',
+    '--output',
+    '.amplify/artifacts/cdk.out',
+  ];
```

Comment:
that should remain in deployer package too.

I.e. deployer should `packageManagerController.runWithPackageManager(commandArgs)` but it should be deployer who composes the `commandArgs`.

Reply:
Done